vcd/resource_vcd_dnat.go

```diff
@@ -220,31 +247,96 @@ func resourceVcdDNATDelete(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	edgeGateway, err := vcdClient.GetEdgeGatewayFromResource(d, "edge_gateway")
+	if err != nil {
+		return fmt.Errorf(errorUnableToFindEdgeGateway, err)
+	}
+
+	if "" != d.Get("network_name").(string) {
+		err = edgeGateway.RemoveNATRule(d.Id())
+		if err != nil {
+			return fmt.Errorf("error deleting SNAT rule: %#v", err)
+		}
+	} else {
+		// this for back compatibility when network name and network type isn't provided - TODO remove with major release
+		task, err := edgeGateway.RemoveNATPortMapping("DNAT",
+			d.Get("external_ip").(string),
+			portString,
+			d.Get("internal_ip").(string),
+			translatedPortString)
+		if err != nil {
+			return fmt.Errorf("error setting DNAT rules: %#v", err)
+		}
+
+		err = task.WaitTaskCompletion()
+		if err != nil {
+			return fmt.Errorf("error completing tasks: %#v", err)
+		}
+	}
+	return nil
+}
+
+func resourceVcdDNATUpdate(d *schema.ResourceData, meta interface{}) error {
 
+	vcdClient := meta.(*VCDClient)
+
+	// Update supports only when network name and network type provided
+	networkName := d.Get("network_name")
+	if nil == networkName || networkName.(string) == "" {
```

Comment:
[Yoda notation](https://en.wikipedia.org/wiki/Yoda_conditions) is not encouraged in Go. Besides, it should be consistent : either always `"literal" == variable` or always `variable == "literal"`. The second one is recommended.
If you run staticcheck, it will show you all the occurrences that should be addressed.
```
resource_vcd_dnat.go:114:5: don't use Yoda conditions (ST1017)
resource_vcd_dnat.go:114:26: don't use Yoda conditions (ST1017)
resource_vcd_dnat.go:128:12: don't use Yoda conditions (ST1017)
resource_vcd_dnat.go:128:33: don't use Yoda conditions (ST1017)
resource_vcd_dnat.go:160:5: don't use Yoda conditions (ST1017)
resource_vcd_dnat.go:254:5: don't use Yoda conditions (ST1017)
resource_vcd_external_network.go:240:45: don't use Yoda conditions (ST1017)
resource_vcd_external_network.go:244:45: don't use Yoda conditions (ST1017)
resource_vcd_external_network.go:248:51: don't use Yoda conditions (ST1017)
resource_vcd_independent_disk.go:192:5: don't use Yoda conditions (ST1017)
resource_vcd_independent_disk.go:201:5: don't use Yoda conditions (ST1017)
resource_vcd_snat.go:78:5: don't use Yoda conditions (ST1017)
resource_vcd_snat.go:78:26: don't use Yoda conditions (ST1017)
resource_vcd_snat.go:89:12: don't use Yoda conditions (ST1017)
resource_vcd_snat.go:89:33: don't use Yoda conditions (ST1017)
resource_vcd_snat.go:115:5: don't use Yoda conditions (ST1017)
resource_vcd_snat.go:190:5: don't use Yoda conditions (ST1017)
resource_vcd_vapp_vm.go:460:33: don't use Yoda conditions (ST1017)
```

Reply:
Fixed for this PR context